packages/next/client/head-manager.ts

```diff
@@ -58,6 +58,9 @@ function updateElements(
       if (title !== document.title) document.title = title
       return
     }
+    if (typeof tag.type === 'function') {
+      return
+    }
```

Comment:
So if a class/functional component is passed as a tag, it breaks `reactElementToDOM`. I just went ahead and did an early return here to avoid the issue.

This is ok the second time around when `updateHead` fires off because I guess `head` is a valid `JSX.Element[]` and isn't a function.